P1_BMI_Stats.R

```diff
@@ -620,136 +651,435 @@ hb_pop_estimates <- hb_pop_estimates %>%
+simd_pop_estimates <- left_join(simd_pop_estimates_1, simd_pop_estimates_2,
+                                by = c("simd" , "schlyr_exam"))
+
+
+
+# create totals for simd (simd_data)
+# all participating boards
+simd_data <- rbind(bmi_basefile %>% group_by(simd, schlyr_exam) %>%
+                     summarise_at(vars(tot:clin_cent_grp7), sum)  %>% ungroup())
+
+# add simd_data to simd_population_estimates
+simd_data <- left_join(simd_data, simd_pop_estimates, 
+                       by = c("simd", "schlyr_exam"))
+
+
+# Confidence intervals (simd)
+# use the function to calculate confidence intervals
+calculate_ci(simd_data)
 
 
+# save as csv file
+write_csv(simd_data, paste0(host_folder, Output, "simd_data.csv"))
 
 
+### data completeness
 
+# calculate scotland population estimates
+sco_pop_estimates <- readRDS(paste0(
+  lookupFolder, "/Unicode/Populations/Estimates/HB2019_pop_est_1981_2018.rds")) %>%
+  rename(year = Year, age = Age, pop = Pop)
 
+sco_pop_estimates <- sco_pop_estimates %>% 
+  filter(age == 5) %>%
+  filter(year >= 2001 & year <=2017) %>% 
+  mutate(schlyr_exam = case_when(year == 2001 ~ "0102", 
+                                 year == 2002 ~ "0203",
+                                 year == 2003 ~ "0304",
+                                 year == 2004 ~ "0405",
+                                 year == 2005 ~ "0506",
+                                 year == 2006 ~ "0607",
+                                 year == 2007 ~ "0708",
+                                 year == 2008 ~ "0809",
+                                 year == 2009 ~ "0910",
+                                 year == 2010 ~ "1011",
+                                 year == 2011 ~ "1112",
+                                 year == 2012 ~ "1213",
+                                 year == 2013 ~ "1314",
+                                 year == 2014 ~ "1415",
+                                 year == 2015 ~ "1516",
+                                 year == 2016 ~ "1617",
+                                 year == 2017 ~ "1718"))  
+
+
+# Scotland level population estimates by year
+sco_pop_estimates <- rbind(sco_pop_estimates %>% 
+                             group_by(schlyr_exam) %>%
+                             summarise(pop = sum(pop)) %>%
+                             mutate(HB2019 = "Total") %>% ungroup())
+
+# save as excel file
+write_csv(sco_pop_estimates, paste0(host_folder, Output,
+                                       "scotland_pop.csv"))
+
+
+
+# board level completeness
+bmi_data_coverage <- readRDS(paste0(host_folder, "bmi_data_coverage.rds"))
+
+# create totals for the number of records with and without a valid height and 
+# weight for individual hb and all participating boards (for bmi_data_coverage)
+# create a variable for the total number of reviews
+bmi_data_coverage <- bmi_data_coverage %>% mutate(count = 1)
+# board level
+hb_p1rev_data <- rbind(bmi_data_coverage %>% group_by(HB2019, schlyr_exam) %>%
```

Comment:
Could do total and valid review sums at the same time but works ok as is.

Reply:
will leave this one for now but worth noting this can be shortened.